src/components/SourcesTree.js

```diff
@@ -41,15 +41,15 @@ let SourcesTree = React.createClass({
     return createTree(this.props.sources);
   },
 
-  queueUpdate: throttle(function() {
+  queueUpdate: debounce(function() {
     if (!this.isMounted()) {
       return;
     }
 
     this.forceUpdate();
-  }, 50),
+  }, 100, { maxWait: 250, leading: false, trailing: true }),
```

Comment:
This means that we debounce for all requests happening within 100 of each other. However we'll never go more than 250 before running the `forceUpdate`.  So this becomes the combination of throttle and debounce, hopefully for the experience we want.